include/xcshell/error_handling.h

```diff
@@ -0,0 +1,16 @@
+#ifndef INCLUDE_XCSHELL_ERROR_HANDLING_H_
+#define INCLUDE_XCSHELL_ERROR_HANDLING_H_
+
+#include <iostream>
+
+class ErrorHandling {
+ private:
+  static void SystemCallNoExitOnFailed();
+  static void SystemCallExitOnFailed();
+
+ public:
+  static void PrintSystemError(std::ostream& os_err);
+  static int ErrorSelector(int return_value, int error_type);
```

Comment:
Looks `ErrorSelector` is not so meaningful, do we have a better name?

Reply:
Can we use other type instead of `int` as error type? For example, We can add a enum class inside class `ErrorHandling`.